tests/countries/test_equatorial_guinea.py

```diff
@@ -0,0 +1,171 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.countries.equatorial_guinea import EquatorialGuinea, GQ, GNQ
+from tests.common import CommonCountryTests
+
+
+class TestEquatorialGuinea(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass(EquatorialGuinea)
+
+    def test_country_aliases(self):
+        self.assertAliases(EquatorialGuinea, GQ, GNQ)
+
+    def test_new_years_day(self):
+        name = "Año Nuevo"
+        self.assertHolidayName(name, (f"{year}-01-01" for year in range(1968, 2050)))
+
+        obs_dt = "2017-01-02"
+        self.assertHolidayName(f"{name} (observado)", obs_dt)
+        self.assertNoNonObservedHoliday(obs_dt)
+
+    def test_womens_day(self):
+        name = "Día Internacional de la Mujer"
+        self.assertHolidayName(name, (f"{year}-03-08" for year in range(1968, 2050)))
+
+    def test_good_friday(self):
+        name = "Viernes Santo"
+        self.assertHolidayName(
+            name,
+            "1994-04-01",
+            "1995-04-14",
+            "1996-04-05",
+            "1997-03-28",
+            "2017-04-14",
+            "2018-03-30",
+            "2019-04-19",
+            "2020-04-10",
+        )
```

Comment:
For movable holidays i..e this and `test_corpus_christi_day`, a general check if it is observed for every year is needed as well:
```suggestion
    def test_good_friday(self):
        name = "Viernes Santo"
        self.assertHolidayName(
            name,
            "1994-04-01",
            "1995-04-14",
            "1996-04-05",
            "1997-03-28",
            "2017-04-14",
            "2018-03-30",
            "2019-04-19",
            "2020-04-10",
        )
        self.assertHolidayName(name, range(1969, 2050))
```

Reply:
It's also reasonable to unify the years interval in movable holidays checks to 2020-2025.